src/gg.regression.unity.bots/Runtime/Scripts/ActionManager/RGGameAction.cs

```diff
@@ -69,6 +72,53 @@ public virtual bool IsEquivalentTo(RGGameAction other)
         {
             return ObjectType == other.ObjectType && ParameterRange.RangeEquals(other.ParameterRange);
         }
+
+        /// <summary>
+        /// Serializes this action into a storable data structure.
+        /// </summary>
+        /// <returns>The serialized action data.</returns>
+        public RGSerializedAction Serialize()
+        {
+            RGSerializedAction result = new RGSerializedAction();
+            result.actionTypeName = GetType().AssemblyQualifiedName;
+            result.paths = new List<string>(Paths.Select(path => string.Join("/", path)));
+            result.objectTypeName = ObjectType.AssemblyQualifiedName;
+            Serialize(result);
+            return result;
+        }
+
+        /// <summary>
+        /// Serialize the action-specific parameters (such as key code, button name, etc.).
+        /// This is stored to the actionParameters field of RGSerializedAction.
+        /// </summary>
+        /// <param name="actionParametersOut">List where the action-specific data should be placed.</param>
+        protected abstract void Serialize(RGSerializedAction serializedAction);
+    }
+    
+    [Serializable]
+    public class RGSerializedAction
+    {
+        public string actionTypeName;
+        public List<string> paths;
+        public string objectTypeName;
+
+        // Used by actions that have a serializable function parameter (e.g. key function, axis name function, etc.)
+        public ActionParamFuncType actionFuncType;
+        public string actionFuncData;
+        
+        // Field where arbitrary string data can be stored
+        public string actionStringData;
+
+        public RGGameAction Deserialize()
```

Comment:
This works, but is inconsistent with how we are doing things in the code in other places.  Everywhere else, we are using newtonsoft json converters to serialize and deserialize objects.  Using newtonsoft everywhere has provided consistency and also allowed us to be able to to versioned deserialization more easily when needed.

Reply:
I've updated this to use Newtonsoft JSON converters, as well as do serialization via string builders in a similar manner as the other objects.